### ISSUE TITLE
request for encoding data as return?

### ISSUE THREAD

**Issue by fayte0618** [tanersener/mobile-ffmpeg#14]
I love your library but it lacks return data aside from success/fail. Can i request that i can listen to the encoding data being output by ffmpeg while running?  I would love to use it for a progress bar.  Thanks!

**Comment by tanersener** [comment]
Library imitates command line usage of FFmpeg most of the time. This is why it doesn't have a data callback functionality.

I'd like to add this feature if it is possible. Do you really need the encoded data? How about an API similar to [cli-wrapper ProgressListener](https://github.com/bramp/ffmpeg-cli-wrapper#get-progress-while-encoding)?

**Comment by fayte0618** [comment]
For UX purposes with the project im making it helps a lot. something similar to the cli-wrapper Progress listener would be great.

**Comment by tanersener** [comment]
OK, I'm working on it. I'll keep you updated.

**Comment by tanersener** [comment]
In dev-v2.x branch, a new callback named `StatsCallback` is introduced for Android. It can be used to listen statistics about the ongoing operation. It has the following fields. Unfortunately, calculating percentage completed was not possible without knowing the total duration & frame number count. But if these values are known, it is possible to use `time` or `videoFrameNumber` values to calculate it in the listener. Are these fields sufficient for your project?

```
    int videoFrameNumber;
    float videoFps;
    float videoQuality;
    long size;
    int time;
    double bitrate;
    double speed;
```
IOS version will have `StatsDelegate` for the same functionality.

**Comment by tanersener** [comment]
Implemented in `v2.1`.
